rules/ruleset.go

```diff
@@ -441,12 +445,43 @@ type RuleSet interface {
 	CutoverNanos() int64
 
 	// TombStoned returns whether the ruleset is tombstoned.
-	TombStoned() bool
+	Tombstoned() bool
 
 	// ActiveSet returns the active ruleset at a given time.
 	ActiveSet(timeNanos int64) Matcher
 
+	// Schema returns the schema.Ruleset representation of this ruleset.
 	Schema() (*schema.RuleSet, error)
+
+	// Tombstone tombstones this ruleset and all of its rules.
+	Tombstone(time.Duration) error
+
+	// AppendMappingRule creates a new mapping rule and adds it to this ruleset.
+	AddMappingRule(string, map[string]string, []policy.Policy, time.Duration) error
```

Comment:
Couple of suggestions:
- Use a struct rather than these long arg lists. It makes it a lot easier for users of the API to know what the method expects.
- Treat `RuleSet` as immutable by having these methods return a new RuleSet after each modification, e.g.`AddMappingRule( ... ) (RuleSet, error)`. This lets people know that original object is unchanged (i.e. still valid if the original was valid), regardless of if operation errors out.